File: resource-registry/src/test/java/nl/haploid/octowight/registry/repository/ResourceElementDmoRepositoryIT.java
package nl.haploid.octowight.registry.repository;

import nl.haploid.octowight.registry.AbstractIT;
import nl.haploid.octowight.registry.TestData;
import org.junit.Test;
import org.springframework.beans.factory.annotation.Autowired;

import java.util.Collections;
import java.util.List;

import static org.junit.Assert.assertEquals;
import static org.junit.Assert.assertNotNull;
import static org.junit.Assert.assertNull;

public class ResourceElementDmoRepositoryIT extends AbstractIT {

	@Autowired
	private ResourceElementDmoRepository resourceElementDmoRepository;

	@Test
	public void testFindByAtomIdAndAtomTypeAndAtomOrigin() {
		final ResourceElementDmo dmo = TestData.resourceElementDmo();
		final ResourceElementDmo expectedDmo = resourceElementDmoRepository.save(dmo);
		final ResourceElementDmo actualDmo = resourceElementDmoRepository
				.findByAtomIdAndAtomTypeAndAtomOrigin(expectedDmo.getAtomId(), expectedDmo.getAtomType(), expectedDmo.getAtomOrigin());
		assertEquals(expectedDmo, actualDmo);
	}

	@Test
	public void testFindByAtomIdInAndAtomTypeAndAtomOrigin() {
		final ResourceElementDmo dmo = TestData.resourceElementDmo();
		final ResourceElementDmo expectedDmo = resourceElementDmoRepository.save(dmo);
		final List<ResourceElementDmo> actualDmos = resourceElementDmoRepository
				.findByAtomIdInAndAtomTypeAndAtomOrigin(Collections.singletonList(expectedDmo.getAtomId()), expectedDmo.getAtomType(), expectedDmo.getAtomOrigin());
		assertEquals(1, actualDmos.size());
		assertEquals(expectedDmo, actualDmos.get(0));
	}

	@Test
	public void testDeleteByResourceTypeAndResourceId() {
		final ResourceElementDmo dmo = TestData.resourceElementDmo();
		final ResourceElementDmo expectedDmo = resourceElementDmoRepository.save(dmo);
		final ResourceElementDmo actualDmoBeforeDelete = resourceElementDmoRepository.findOne(expectedDmo.getId());
		assertNotNull(actualDmoBeforeDelete);
		resourceElementDmoRepository.deleteByResourceTypeAndResourceId(expectedDmo.getResourceType(), expectedDmo.getResourceId());
		final ResourceElementDmo actualDmoAfterDelete = resourceElementDmoRepository.findOne(expectedDmo.getId());
		assertNull(actualDmoAfterDelete);
	}
}

File: resource-registry/src/test/scala/nl/haploid/octowight/registry/repository/ResourceElementDmoRepositoryIT.scala
package nl.haploid.octowight.registry.repository

import java.util

import nl.haploid.octowight.registry.{AbstractIT, TestData}
import org.springframework.beans.factory.annotation.Autowired

class ResourceElementDmoRepositoryIT extends AbstractIT {
  @Autowired private[this] val resourceElementDmoRepository: ResourceElementDmoRepository = null

  behavior of "Resource element repository"

  it should "find by atom id, type & origin" in {
    val dmo = TestData.resourceElementDmo
    val expectedDmo = resourceElementDmoRepository.save(dmo)
    val actualDmo = resourceElementDmoRepository.findByAtomIdAndAtomCategoryAndAtomOrigin(
      expectedDmo.atom.id, expectedDmo.atom.category, expectedDmo.atom.origin)
    actualDmo should be(expectedDmo)
  }

  it should "find by atom ids, type & origin" in {
    val dmo = TestData.resourceElementDmo
    val expectedDmo = resourceElementDmoRepository.save(dmo)
    val actualDmos = resourceElementDmoRepository.findByAtomIdInAndAtomCategoryAndAtomOrigin(
      util.Collections.singletonList(expectedDmo.atom.id), expectedDmo.atom.category, expectedDmo.atom.origin)
    actualDmos should have size 1
    actualDmos.get(0) should be(expectedDmo)
  }

  it should "delete by resource id & collection" in {
    val dmo = TestData.resourceElementDmo
    val expectedDmo = resourceElementDmoRepository.save(dmo)
    val actualDmoBeforeDelete = resourceElementDmoRepository.findOne(expectedDmo.id)
    actualDmoBeforeDelete should not be null
    resourceElementDmoRepository.deleteByResourceCollectionAndResourceId(expectedDmo.resourceCollection, expectedDmo.resourceId)
    val actualDmoAfterDelete = resourceElementDmoRepository.findOne(expectedDmo.id)
    actualDmoAfterDelete should be(null)
  }
}
